service/history/transfer_queue_active_task_executor.go

```diff
@@ -908,6 +946,76 @@ func (t *transferQueueActiveTaskExecutor) processStartChildExecution(
 	}, parentClock, childClock)
 }
 
+// verifyChildWorkflow describes the childWorkflowID and identifies its parent. It then checks if the current run was derived from that parent by comparing the OriginalRunID value.
+// It returns the child's runID if the checks pass. Empty runID is returned if the check doesn't pass.
+func (t *transferQueueActiveTaskExecutor) verifyChildWorkflow(
+	ctx context.Context,
+	// task *tasks.ResetWorkflowTask,
+	mutableState workflow.MutableState,
+	childNamespace *namespace.Namespace,
+	childWorkflowID string,
+) (childID string, retError error) {
+	childDescribeReq := &historyservice.DescribeWorkflowExecutionRequest{
+		NamespaceId: childNamespace.ID().String(),
+		Request: &workflowservice.DescribeWorkflowExecutionRequest{
+			Namespace: childNamespace.Name().String(),
+			Execution: &commonpb.WorkflowExecution{
+				WorkflowId: childWorkflowID,
+			},
+		},
+	}
+	response, err := t.historyRawClient.DescribeWorkflowExecution(ctx, childDescribeReq)
+	if err != nil {
+		// It's not an error if the child is not found. Return empty childID so that the child is created.
+		if common.IsNotFoundError(err) {
+			return "", nil
+		}
+		return "", err
+	}
+
+	if response.WorkflowExecutionInfo.ParentExecution == nil {
+		// The child doesn't have a parent. Maybe it was started by some client.
+		return "", nil
+	}
+	// Verify if the WorkflowIDs match first.
+	if response.WorkflowExecutionInfo.ParentExecution.WorkflowId != mutableState.GetExecutionInfo().WorkflowId {
+		return "", nil
+	}
+
+	childsParentRunID := response.WorkflowExecutionInfo.ParentExecution.RunId
+	// Check if the child's parent was the base run for the current run.
+	if childsParentRunID == mutableState.GetExecutionInfo().OriginalExecutionRunId {
+		return response.WorkflowExecutionInfo.Execution.RunId, nil
+	}
+
+	// load the child's parent mutable state.
```

Comment:
We could consider storing parent original runID in child's mutable state as well. but probably not necessary for this low volume case

Reply:
Agree.